nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -1189,4 +1180,17 @@ public List<ConfigVerificationResult> verify(final ProcessContext context, final
 
         return results;
     }
+
+    /**
+     * Return whether to include the entity in the listing, based on the minimum and maximum object age (if configured).
+     */
+    private boolean includeObjectInListing(final ProcessContext context, final S3VersionSummary versionSummary) {
+        final Long minAgeMilliseconds = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+        final Long maxAgeMilliseconds = context.getProperty(MAX_AGE) != null ? context.getProperty(MAX_AGE).asTimePeriod(TimeUnit.MILLISECONDS) : null;
+        final long lastModifiedTime = versionSummary.getLastModified().getTime();
+        final long currentTime = System.currentTimeMillis();
```

Comment:
The current time should be evaluated only once per verification/listing. Otherwise items may or may not be listed depending on the time the filter is executed for the given item (even if 2 items have the same timestamp). So `currentTime` should be a parameter of this method.